functions/src/definitions/eventHandlers/checkerHandlerTelegram.ts

```diff
@@ -146,4 +201,349 @@ const onCheckerPublishTelegram = onMessagePublished(
   }
 )
 
+const sendNamePrompt = async (
+  chatId: number,
+  checkerQuerySnap: QuerySnapshot
+) => {
+  const namePrompt = await bot.sendMessage(
+    chatId,
+    "Hi, what's your full name?",
+    {
+      reply_markup: {
+        force_reply: true,
+      },
+    }
+  )
+
+  bot.onReplyToMessage(chatId, namePrompt.message_id, async (nameMsg) => {
```

Comment:
Unfortunately, even though this works in the emulator, this will not work in the deployed serverless environment. Becauase by the time the user replies on telegram, the instance which attached the callback to the bot object might not exist already. See the last few messages in https://chatgpt.com/share/6cc788a7-521a-459a-bcb7-9329b121d691. Basically, we'll need to store the message IDs of the name prompt, the number prompt, and the otp prompt inside the DB. And then when a new forceReply message comes in, check against that to find out which stage it is at.